### PR TITLE
feat: Improve ceremony metrics

### PR DESCRIPTION
**Pull request by marcellorigotti**
# Pull Request

Closes: PLA-1021

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [x] I have updated documentation where appropriate.

## Summary

- Removed ceremony_id and the Drop implementation from the metrics (which now don't need to be deleted anymore)
- Introduced histogram instead which allows us to track different buckets of values (ceremonies/stages duration) 
- Added pending_ceremony metric to keep track of the number or running ceremonies

### CODE REVIEW

**Comment by marcellorigotti** [comment]
TODO: decide the bucket ranges for ceremony_duration and stage_duration

File: utilities/src/with_std/metrics.rs [hunk 1]
```diff
@@ -164,6 +192,70 @@ macro_rules! build_counter_vec {
...
+				metric: &'static $metric_ident,
+				const_labels: [String; { $const_labels.len() }],
+			) -> $struct_ident {
+				$struct_ident { metric, const_labels }
+			}
+
+			pub fn observe(
+				&self,
+				non_const_labels: &[&str; { $labels.len() - $const_labels.len() }],
+				val: u128,
```
Comment (AlastairHolmes) [inline utilities/src/with_std/metrics.rs:230]: This could take a std::time::Duration

File: utilities/src/with_std/metrics.rs [hunk 2]
```diff
@@ -455,50 +547,50 @@ build_counter_vec!(
...
-	StageDurationDrop,
+	StageDuration,
 	"stage_duration",
 	"Measure the duration of a stage in ms",
-	true,
-	["chain", "ceremony_id", "stage", "phase"], //phase can be either receiving or processing
-	["chain", "ceremony_id"]
+	["chain", "stage", "phase"], //phase can be either receiving or processing
+	["chain"],
+	(vec![1.0, 2.0, 3.0])
```
Comment (AlastairHolmes) [inline utilities/src/with_std/metrics.rs:487]: Lets change these to be in seconds, and use 4, 8, 16, 32, 64, 128, 256, 512, 1024, for both Histograms

File: engine/multisig/src/client/ceremony_manager.rs [hunk 3]
```diff
@@ -416,6 +416,10 @@ impl<Chain: ChainSigning> CeremonyManager<Chain> {
 				}
 			},
 		}
+		PENDING_CEREMONIES
```
Comment (msgmaxim) [inline engine/multisig/src/client/ceremony_manager.rs:419]: We can put these into the corresponding signing/keygen branches above, right?

**Review by msgmaxim — APPROVED** [review]

File: utilities/src/with_std/metrics.rs [hunk 4]
```diff
@@ -624,15 +624,15 @@ mod test {
...
 					metrics.stage_completing.inc(&["stage1"]);
 					metrics.stage_completing.inc(&["stage1"]);
 					metrics.stage_completing.inc(&["stage2"]);
-					metrics.stage_duration.observe(&["stage1", "receiving"], 780);
-					metrics.stage_duration.observe(&["stage1", "processing"], 78);
+					metrics.stage_duration.observe(&["stage1", "receiving"], Duration::new(780, 0));
+					metrics.stage_duration.observe(&["stage1", "processing"], Duration::new(78, 0));
 					metrics.stage_failing.inc(&["stage3", "NotEnoughMessages"]);
 
 					//This request does nothing, the ceremony is still ongoning so there is no deletion
```
Comment (AlastairHolmes) [inline utilities/src/with_std/metrics.rs:638]: Below can we update: "# HELP ceremony_duration Measure the duration of a ceremony in ms" to seconds.

Comment (AlastairHolmes) [reply utilities/src/with_std/metrics.rs:638]: And the metric describes of course

**Review by AlastairHolmes — APPROVED** [review]

**Comment by marcellorigotti** [comment]
Renaming done, once CI passes we can merge!

**Pull request merged**